DemocracyClub/EveryElection
# Feat/admin panel updates

This PR combines 5 small tasks on the admin panel - 
[Validate seats contested <= seats total](https://app.asana.com/0/1204880927741389/1205231021592931/f)
[Snooped election should have a linked model instead of a select field](https://app.asana.com/0/1204880927741389/1205231021592932/f)
[Hide tmp election ID from the admin panel](https://app.asana.com/0/1204880927741389/1205231021592933/f)
[Rename group as "parent" in the admin frontend](https://app.asana.com/0/1204880927741389/1205231021592934/f)
[Where group type is null, substitute the word "ballot"](https://app.asana.com/0/1204880927741389/1205231021592935/f)

It also addresses a missed black check from the previous PR and removes node install from the CircleCI pipeline because it was causing problems. 

Sorry there's a lot of different things going on, but I thought the best chance of this getting merged before I leave was to roll these little bits together 🗞️ 

<img width="503" alt="Screenshot 2023-11-01 at 18 11 39" src="https://github.com/DemocracyClub/EveryElection/assets/9531063/1369382a-41ee-4603-b3d6-655da80eb730">

<img width="731" alt="Screenshot 2023-11-01 at 18 27 51" src="https://github.com/DemocracyClub/EveryElection/assets/9531063/bc3eef06-cac8-463a-9ba2-f0fb80cbfa03">


To test:
1. Find any election with a seats total value. Try updating the seats contested value to be greater than seats total. You should see a validation error. 
2. Find any election which has a snooped election (I used election ID `10895` for this). In the Source section you should see a clickable object. If you click though, the snooped election page shouldn't allow creations, edits or deletes.
3. Go to any election. Expand the metadata section. There shouldn't be a key for "tmp election id"
4. Go to any election. In the first section you should see a key for "Parent" and should no longer see a key for "Group"
5. Go to any ballot-level election. You should see "ballot" next to group type.

```[tasklist]
### PR Checklist
```
